tools/halide_image.h

```diff
@@ -554,38 +554,38 @@ struct for_each_element_helpers {
      * resolution. The decltype is to make this version impossible if
      * the function is not callable with this many args. */
     template<typename ...Args>
-    static auto for_each_element_variadic(int, int d, Fn f, const buffer_t &buf, const Args &... args)
```

Comment:
Yeah, I think the for_each helpers should be reverted to Args...

This file is all changing anyway. I'm currently fighting the android compiler on these, so I'm messing with this code quite a bit in another branch.